File: JAVA/Lecturer.java
// Lecturer mewarisi Person (Hierarchical Inheritance)
public class Lecturer extends Person {
    private String nidn;
    private String departemen;
    private String jabatan;

    public Lecturer() {
        super();
        this.nidn = "";
        this.departemen = "";
        this.jabatan = "";
    }

    public Lecturer(String id, String nama, int usia, String nidn, String departemen, String jabatan) {
        super(id, nama, usia);
        this.nidn = nidn;
        this.departemen = departemen;
        this.jabatan = jabatan;
    }

    // Getter & Setter
    public String getNidn() { return nidn; }
    public void setNidn(String nidn) { this.nidn = nidn; }

    public String getDepartemen() { return departemen; }
    public void setDepartemen(String departemen) { this.departemen = departemen; }

    public String getJabatan() { return jabatan; }
    public void setJabatan(String jabatan) { this.jabatan = jabatan; }
}

File: CPP/Lecturer.cpp
#pragma once
#include "Person.cpp"

// Lecturer mewarisi Person (hierarchical inheritance)
class Lecturer : public Person {
private:
    string nidn;
    string departemen;
    string jabatan;

public:
    Lecturer() : Person(), nidn(""), departemen(""), jabatan("") {}
    Lecturer(string id, string nama, int usia, string nidn, string departemen, string jabatan)
        : Person(id, nama, usia), nidn(nidn), departemen(departemen), jabatan(jabatan) {}

    // Getter
    string getNidn() const { return nidn; }
    string getDepartemen() const { return departemen; }
    string getJabatan() const { return jabatan; }

    // Setter
    void setNidn(string nidn) { this->nidn = nidn; }
    void setDepartemen(string departemen) { this->departemen = departemen; }
    void setJabatan(string jabatan) { this->jabatan = jabatan; }
};